File: pyjl/tests/tests_new/dataclass_test.py
from dataclasses import dataclass

@jl_dataclass # Remove when testing
@dataclass(init=True, eq=True, order=True, unsafe_hash=False, frozen=False)
class ValueHolder:
    val: int
    strVal: str

if __name__ == "__main__":
    a = ValueHolder(10, "1")
    print(a)
    a.__init__(2, "10")
    print(a)

File: pyjl/tests/tests_new-py2many/dataclass_test.jl
using DataClass

@dataclass mutable struct ValueHolder
val::Int64
strVal::String
_initvars = [_init=true, _repr=true, _eq=true, _order=false, _unsafe_hash=false, _frozen=false]
end

@dataclass mutable struct Item
id::String
price_per_unit::Float64
quantity::Int64
_initvars = [_init=true, _repr=true, _eq=true, _order=false, _unsafe_hash=false, _frozen=false]
end

function main()
a = ValueHolder(10, "1")
println(a);
__init__(a, 2, "10");
println(a);
end

main()
